fable/src/fable/version_test.cpp

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2023 Robert Bosch GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <fable/version.hpp>
+
+TEST(fable_version, check_valid) {
+  ASSERT_NE(FABLE_VERSION_U32, 0);
```

Comment:
i feel like a test where the version is not 0.0.0 and you also check for major minor patch may be more meaningful

Reply:
I'm not sure how such a test would be stable, since that value is coming from the actual version of fable. It would break as soon as I make `fable 1.0`. I would have to set the version manually, which proves nothing.

Reply:
i think we can trust that configure_file does the right thing. you could (eg) define the u32 version before including the header to mock a different version than 0.0.0